cycles-ledger/tests/tests.rs

```diff
@@ -997,6 +1143,64 @@ fn test_deduplication() {
     .unwrap();
 }
 
+#[test]
+fn test_deduplication_with_insufficient_funds() {
```

Comment:
Essentially, the test ensures that the right error message is returned (duplicate error rather than insufficient funds error), right?
I think this is not immediately clear to the reader. Could you add a comment to clarify this?